code/datums/components/immersion.dm

```diff
@@ -0,0 +1,84 @@
+#define EXHALE 0
+#define INHALE 1
+
+
+/datum/component/immersion
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+
+	var/obj/item/organ/eyes/E
+	var/obj/item/organ/lungs/L
+	var/mob/living/carbon/C
+	var/last_blink
+	var/last_breath
+	var/blink_every = 24 SECONDS
+	var/breathe_every = 14 SECONDS
+	var/next_breath_type = INHALE
+	var/warned_blink = FALSE
+	var/warned_breath = FALSE
+
+/datum/component/immersion/Initialize()
+	if(!iscarbon(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	C = parent
+	E = C.getorganslot(ORGAN_SLOT_EYES)
+	L = C.getorganslot(ORGAN_SLOT_LUNGS)
```

Comment:
These don't update should parent get their organs switched out. Either add signals listening to organ insertion/removal and then update or just get them every tick.

Reply:
Also clear them on destroy. Just a good habit to have.